stan/math/prim/scal.hpp

```diff
@@ -78,6 +79,7 @@
 #include <stan/math/prim/scal/fun/identity_free.hpp>
 #include <stan/math/prim/scal/fun/if_else.hpp>
 #include <stan/math/prim/scal/fun/inc_beta.hpp>
+#include <stan/math/prim/scal/fun/init_threadpool_tbb.hpp>
```

Comment:
Is there a reason these sit in prim/scal/fun? Maybe we should have a `backend` folder where we put both TBB and OpenCL stuff

Reply:
I am open to suggestions. Another option could be `stan/math/core` which is motivated by `rev/core` - but I am really open to where to stick this; I thought `prim/scal/fun` reflects the right dependency which is why I put it there.

Reply:
I think core is fine. I imagine there wont be much "TBB core" code outside of this which would justify making another folder. Other TBB code is going to live either inside of Stan code or specific stan math functions right?

Reply:
Either way this is a review-time detail.

Reply:
The `map_rect` which uses the TBB will, of course, stay where it is. So not much other TBB code (if any) will go into `stan/math/core`. It just needs to be included always.